src/internal.c

```diff
@@ -8982,28 +8970,24 @@ int SendKexDhReply(WOLFSSH* ssh)
         }
         if (ret == 0) {
             c32toa(ssh->kSz + kPad, scratchLen);
-            ret = HashUpdate(&ssh->handshake->hash, enmhashId,
-                             scratchLen, LENGTH_SZ);
+            ret = HashUpdate(hash, hashId, scratchLen, LENGTH_SZ);
         }
         if ((ret == 0) && (kPad)) {
             scratchLen[0] = 0;
-            ret = HashUpdate(&ssh->handshake->hash,
-                             enmhashId, scratchLen, 1);
+            ret = HashUpdate(hash, hashId, scratchLen, 1);
         }
         if (ret == 0) {
-            ret = HashUpdate(&ssh->handshake->hash, enmhashId,
-                             ssh->k, ssh->kSz);
+            ret = HashUpdate(hash, hashId, ssh->k, ssh->kSz);
         }
 
         /* Save the exchange hash value H, and session ID. */
         if (ret == 0) {
-            ret = wc_HashFinal(&ssh->handshake->hash,
-                               enmhashId, ssh->h);
-            wc_HashFree(&ssh->handshake->hash, enmhashId);
-            ssh->handshake->hashId = WC_HASH_TYPE_NONE;
+            ssh->hSz = wc_HashGetDigestSize(hashId);
```

Comment:
This can return a BAD_FUNC_ARG value (-173). It should never happen since earlier 'update' calls were done with the same hashId but we should sanity check the return value before use later in the WMEMCPY call.

Reply:
Fixed.